### DISCUSSION TITLE
useRoute() params empty at the beginning

### DISCUSSION THREAD

**Discussion by arashctl** [nuxt/nuxt#17739] [Questions] [answered]
I have this simple dynamic route `[contentId].vue` in my `pages` directory. I have a `NuxtLink` to `/someId` in my `index.vue`, and `[slug].vue looks like this: 
```
const route = useRoute()
const contentId = route.params.contentId
console.log(contentId)

const { data: pageData, pending: pageDataPending } = await useFetch(() => `/content/page/${contentId}/`, {
    key: `page_${contentId.value}`,
    ...options
})
```

I can't reproduce this in a blank project, but this is already quite minimal. The `console.log` statement prints `undefined` (`useRoute().params is an empty object`), and `useFetch` requests the route `/content/page/undefined/` as well. 

If I replace the contentId with this: 
```
const contentId = computed(() => route.params.contentId)
```
Devtools shows the correct `contentId` value, which is `someId`, but useFetch still requests the undefined path and the `console.log` statement prints undefined. 

Seems like `useRoute().params` is being populated a little bit late!

Could someone please explain what's going on?

**Comment by danielroe** [comment, marked as answer]
What's your Nuxt version?

**Reply by arashctl** [reply]
> What's your Nuxt version?

- Operating System: `Windows_NT`
- Node Version:     `v16.13.2`
- Nuxt Version:     `3.0.0-27486226.7a70644`
- Package Manager:  `yarn@1.22.17`
- Builder:          `vite`
- User Config:      `serverMiddleware`, `privateRuntimeConfig`, `publicRuntimeConfig`, `css`, `buildModules`, `build`
- Runtime Modules:  `-`
- Build Modules:    `@pinia/nuxt@0.1.8`, `@nuxtjs/tailwindcss@5.0.0-4`

**Reply by danielroe** [reply]
Update your Nuxt version and you should be fine 👍

**Reply by arashctl** [reply]
> Update your Nuxt version and you should be fine 👍

Resolved, thanks!
Where can I see the list of releases? I thought I was on the latest version.

**Reply by danielroe** [reply]
Every commit triggers an edge release. When we move into RC, then the edge releases will continue but the RC releases will be less frequent and include changelogs.

**Comment by arashctl** [comment]
@danielroe I updated my Nuxt version, and my problem was solved in dev environment. 
However, after deploying to heroku, my index page is giving me 404 not found, ONLY on ssr mode. 
I can open other pages via SSR, and open the index via `NuxtLink`s in those pages, but index in ssr is giving me 404 not found with no appropriate logs. 

Do you have any idea why that might happen?

![image](https://user-images.githubusercontent.com/38472389/162559917-4b58ca71-19de-4ed9-8194-e3875a9e0ab8.png)

**Reply by danielroe** [reply]
Difficult to say. Would you provide a reproduction?

**Reply by arashctl** [reply]
@danielroe Just found out the problem goes away if I comment this out: 

```
export default defineNuxtConfig({
    serverMiddleware: process.env.REDIRECT_SSL == 'redirect-ssl' ? ['redirect-ssl'] : [],
})
```

I was using this without a problem until yesterday. Maybe new releases are not compatible? I guess it should have something to do with the new nitro PR
